lib/registry.js

```diff
@@ -89,12 +89,11 @@ class Registry {
 				return item;
 			}
 
-			item.values = item.values.map(val =>
-				// Avoid mutation and merge metric labels with registry default labels
-				Object.assign({}, val, {
-					labels: Object.assign({}, this._defaultLabels, val.labels)
-				})
-			);
+			for (const val of item.values) {
+				for (const label of Object.keys(this._defaultLabels)) {
```

Comment:
Since this is a perf PR, we can lift `Object.keys(this._defaultLabels)` put of the inner loop to avoid iterating through that object every time

Reply:
Definitely